File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)

        longest = 0

        for num in nums:
            if num - 1 not in s:
                currLen = 1
                currNum = num + 1

                while currNum in s:
                    currLen += 1
                    currNum += 1
                
                if currLen > longest:
                    longest = currLen
        
        return longest

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        std::unordered_map<int, int> prefixSum;
        prefixSum[0] = 1;
        int currSum = 0;
        int count = 0;

        for (const auto& num : nums) {
            currSum += num;
            int diff = currSum - k;

            if (prefixSum.contains(diff)) {
                count += prefixSum[diff];
            }

            prefixSum[currSum] += 1;
        }

        return count;
    }
};